tools/lkl/lib/fs.c

```diff
@@ -276,6 +276,30 @@ struct lkl_dir *lkl_opendir(const char *path, int *err)
 	return dir;
 }
 
+struct lkl_dir *lkl_fdopendir(int fd, int *err)
+{
+	struct lkl_dir *dir = lkl_host_ops.mem_alloc(sizeof(struct lkl_dir));
+
+	if (!dir) {
+		*err = -LKL_ENOMEM;
+		return NULL;
+	}
+
+	dir->fd = fd;
+
+	dir->len = 0;
+	dir->pos = NULL;
+
+	return dir;
+}
```

Comment:
There is some code we can share between lkl_fdopendir and lkl_opendir. Maybe add a new function lkl_dir_alloc that we can use in both?